File: 1224-maximum-equal-frequency/1224-maximum-equal-frequency.cpp
class Solution {
public:
    int maxEqualFreq(vector<int>& nums) {
        vector<int> count(100001);
        vector<int> freq(100001);
        int n = nums.size();
        for (int num : nums) {
            ++count[num];
            ++freq[count[num]];
        }
        for (int i = n - 1; i > 0; --i) {
            if (count[nums[i]] * freq[count[nums[i]]] == i) return i + 1;
            --freq[count[nums[i]]];
            --count[nums[i]];
            if (count[nums[i - 1]] * freq[count[nums[i - 1]]] == i) return i + 1;
        }
        return 1;

    }
};

File: 1224-maximum-equal-frequency/1224-maximum-equal-frequency.py
class Solution(object):
    def maxEqualFreq(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = [0] * 100001
        freq = [0] * 100001
        n = len(nums)
        
        for num in nums:
            count[num] += 1
            freq[count[num]] += 1
            
        for i in range(n - 1, 0, -1):
            if count[nums[i]] * freq[count[nums[i]]] == i:
                return i + 1
            freq[count[nums[i]]] -= 1
            count[nums[i]] -= 1
            if count[nums[i - 1]] * freq[count[nums[i - 1]]] == i:
                return i + 1
        return 1